From: NixOS Android devices contributors <nixos-android-devices@localhost>
Date: Tue, 29 Sep 2026 00:53:00 +0800
Subject: [PATCH] arm: dts: asphalt: configure volume-up input supply and bias

Port the GPIO6 input-enable, bias-pull-up and power-source=1 state from the
working Linux board DTS, using U-Boot's normal function name. Without this,
button-kbd receives spurious UP input and interrupts unattended EFI startup.

Same board source as the initial DTS adaptation:
https://github.com/lenovo-basements/linux/commit/d792042c0c9561c0f32a8f6eb9f0aca325fc0350
pkgs/linux-kernel/sm8475/asphalt/patches/local/0006-asphalt-board-dts.patch
---
--- a/dts/upstream/src/arm64/qcom/sm8475-lenovo-asphalt.dts
+++ b/dts/upstream/src/arm64/qcom/sm8475-lenovo-asphalt.dts
@@ -62,6 +62,8 @@
 
 	gpio-keys {
 		compatible = "gpio-keys";
+		pinctrl-names = "default";
+		pinctrl-0 = <&key_vol_up_default>;
 		key-volume-up {
 			label = "Volume Up";
 			gpios = <&pm8350_gpios 6 GPIO_ACTIVE_LOW>;
@@ -131,3 +133,13 @@
 		       <75000000 300000000>, <0 0>, <0 0>, <0 0>, <0 0>;
 	status = "okay";
 };
+
+&pm8350_gpios {
+	key_vol_up_default: key-vol-up-state {
+		pins = "gpio6";
+		function = "normal";
+		input-enable;
+		bias-pull-up;
+		power-source = <1>;
+	};
+};
